mixer-ngx/src/ngx/server_config.rs

```diff
@@ -38,53 +38,82 @@ impl MixerConfig for  ngx_http_mixer_srv_conf_t  {
     fn process_istio_attr(&self, attr: &mut AttributeWrapper) {
         ngx_event_debug!("Processing server attributes");
 
-        ngx_event_debug!("{}: {}",DESTINATION_SERVICE, self.destination_service.to_str());
+        let dest_ip = self.destination_ip.to_str();
+        let source_ip = self.source_ip.to_str();
+        if dest_ip.len() != 0 {
+            attr.insert_ip_attribute( DESTINATION_IP, dest_ip.parse::<Ipv4Addr>().unwrap());
+        }
+        if source_ip.len() != 0 {
+            attr.insert_ip_attribute( SOURCE_IP, source_ip.parse::<Ipv4Addr>().unwrap());
+        }
+
         attr.insert_string_attribute( DESTINATION_SERVICE, self.destination_service.to_str());
-        ngx_event_debug!("{}: {}", DESTINATION_UID, self.destination_uid.to_str());
         attr.insert_string_attribute( DESTINATION_UID, self.destination_uid.to_str());
-        ngx_event_debug!("{}: {}",DESTINATION_IP, self.destination_ip.to_str());
-        attr.insert_string_attribute( DESTINATION_IP, self.destination_ip.to_str());
-        ngx_event_debug!("{}: {:?}", DESTINATION_LABELS, labels_to_map(&self.destination_labels));
-        attr.insert_string_map( DESTINATION_LABELS, labels_to_map(&self.destination_labels));
-        let ip = self.source_ip.to_str();
-        if ip.len() != 0 {
-            ngx_event_debug!("{}: {:?}", SOURCE_IP, ip);
-            attr.insert_ip_attribute( SOURCE_IP, ip.parse::<Ipv4Addr>().unwrap());
-        }
-        ngx_event_debug!("{}: {}", SOURCE_UID, self.source_uid.to_str());
+        attr.insert_string_map( DESTINATION_LABELS, self.destination_labels.to_map());
         attr.insert_string_attribute( SOURCE_UID,self.source_uid.to_str());
-        ngx_event_debug!("{}: {}", SOURCE_SERVICE, self.source_service.to_str());
         attr.insert_string_attribute( SOURCE_SERVICE,self.source_service.to_str());
-        ngx_event_debug!("{}: {}", SOURCE_PORT, self.source_port as i64);
         attr.insert_int64_attribute( SOURCE_PORT,self.source_port as i64);
-        ngx_event_debug!("{}: {:?}", SOURCE_LABELS, labels_to_map(&self.source_labels));
-        attr.insert_string_map( SOURCE_LABELS, labels_to_map(&self.source_labels));
-
+        attr.insert_string_map( SOURCE_LABELS, self.source_labels.to_map());
     }
 }
 
-fn labels_to_map(labels: *const service_labels) -> HashMap<String,String> {
+impl service_labels {
 
-    let mut out_map: HashMap<String,String> = HashMap::new();
+    fn to_map(&self) -> HashMap<String,String> {
 
-    unsafe {
-        if ((*labels).key as *const u8).is_null() {
+        let mut out_map: HashMap<String,String> = HashMap::new();
+
+        if (self.key as *const u8).is_null() {
             return out_map;
+        } else {
+            for (key,value) in self.iter() {
```

Comment:
Added iterator for service_labels struct to more clearly add elements to string map.